### DISCUSSION TITLE
How to better tune Zeek to capture C2 Long Polling?

### DISCUSSION THREAD

**Discussion by ColeVan** [Security-Onion-Solutions/securityonion#10577] [Unsupported Versions]
Very interesting 9-hour meterpreter session. This is all testing data inside my CyberLabz home lab environment. I'm trying to better understand how to tune Zeek to quickly capture and highlight initial C2 connections without waiting for the connection to close. It seems that Zeek will present the log in Kibana or "Hunt" once the connection is finally terminated.
 
The connection is initiated around 02:49 UTC and lasted for 9 hours.

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/70167373/20a9b556-4a37-4ab7-87a9-94943dd5d878)

After 9 hours the session is killed and only a handful of connections logs are found. The event duration is very high which could be an indicator. This to me looks like Long Polling which Zeek will log once the connection is closed.
 
This type of C2 communication would be hard to find. Are there any other ways to tune Zeek to better highlight potential long-polling C2 behavior?

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/70167373/f536c58b-23a9-486d-8dbb-3790c30d3deb)

**Comment by dougburks** [comment]
Would something like this help?
https://github.com/corelight/zeek-long-connections

**Reply by ColeVan** [reply]
Hey Doug,

Thanks for sharing this. I'll try this out and report back.
